exp/util/util.go

```diff
@@ -63,7 +63,7 @@ func GetMachinePoolByName(ctx context.Context, c client.Client, namespace, name
 func MachinePoolToInfrastructureMapFunc(gvk schema.GroupVersionKind, log logr.Logger) handler.MapFunc {
 	log = log.WithValues("machine-pool-to-infra-map-func", gvk.String())
 	return func(o client.Object) []reconcile.Request {
-		log := log.WithValues("namespace", o.GetNamespace(), "name", o.GetName())
+		log := log.WithValues("Namespace", o.GetNamespace(), "name", o.GetName())
 		m, ok := o.(*expv1.MachinePool)
 		if !ok {
 			log.V(4).Info("not a machine pool")
```

Comment:
```suggestion
			log.V(4).Info("not a machine pool", "Object", klog.KObj(o))
```
\+ drop l.66

> When Kubernetes object kind is unknown without runtime checking we should use object key.
> https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/migration-to-structured-logging.md#name-arguments

Reply:
\+ Add 
```go
log := log.WithValues("MachinePool", klog.KObj(o))
```
after the if

and drop namespace/name from l.82

Reply:
Regarding the first log message. Maybe we can just drop it. Looked through a bunch of other handler.Funcs and we usually just return nil without log, e.g.:
```go
		cluster, ok := o.(*clusterv1.Cluster)
		if !ok {
			return nil
		}
```

Reply:
I'm going to leave the log there for now, but I've made the other changes.